File: src/Keyboard.fs
[<RequireQualifiedAccess>]
module Keyboard

open Fable.Core
open Browser.Types

[<ImportMember("./Interop/Keyboard.js")>]
let bindToContext
  (bindings: (string * (KeyboardEvent -> unit)) array)
  (ctx: string)
  : unit =
  jsNative

[<ImportMember("./Interop/Keyboard.js")>]
let bindSingleToGlobal
  (keyboardSequence: string)
  (callback: (KeyboardEvent -> unit))
  : unit =
  jsNative

[<ImportMember("./Interop/Keyboard.js")>]
let start () : unit = jsNative

[<ImportMember("./Interop/Keyboard.js")>]
let stop () : unit = jsNative

[<ImportMember("./Interop/Keyboard.js")>]
let setContext (context: string) : unit = jsNative

File: src/Interop/Keyboard.js
import keyboardjs from 'keyboardjs';

export function bindToContext(bindings = [], ctx = 'global') {
    keyboardjs.setContext(ctx);
    for (const [keySeqyence, callback] of bindings) {
        keyboardjs.bind(keySeqyence, callback, null, true);
    }
    keyboardjs.setContext('global');
}

export function bindSingleToGlobal(keyboardSequence, callback) {
    const binding = [keyboardSequence, callback];
    bindToContext([binding]);
}

export function start() {
    keyboardjs.watch();
}

export function stop() {
    keyboardjs.stop();
}

export function setContext(context = 'global') {
    keyboardjs.setContext(context);
}